src/utils/StreamHighlightUtils.ts

```diff
@@ -77,13 +77,21 @@ export class StreamHighlightUtils {
 
 export function getRestHighlightsForAllTerms(
   toHighlight: string,
-  termsToHighlight: { [originalTerm: string]: string[] },
-  phrasesToHighlight: { [phrase: string]: { [originalTerm: string]: string[] } },
+  termsToHighlight: IHighlightTerm,
+  phrasesToHighlight: IHighlightPhrase,
   opts: IStreamHighlightOptions
 ): IHighlight[] {
   const indexes = [];
   const sortedTerms = _.keys(termsToHighlight).sort(termsSorting);
-  _.each(sortedTerms, (term: string) => {
+
+  const phrasesTerms = _.chain(phrasesToHighlight)
+    .values()
+    .map(_.keys)
+    .flatten()
+    .value();
+  const uniqueTerms = _.difference(sortedTerms, phrasesTerms);
+
+  _.each(uniqueTerms, (term: string) => {
```

Comment:
It would be nice to extract all the code needed to generate the array of unique terms into a separate method. e.g.
`private getUniqueTerms(termsToHighlight, phrasesToHighlight): string[] {}`

Reply:
Done